src/backend/db/migrations/1_initial-migrations.sql

```diff
@@ -4,7 +4,32 @@ CREATE TABLE "user" (
   user_id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
   first_name TEXT NOT NULL,
   last_name TEXT NOT NULL,
-  role TEXT NOT NULL CHECK (role IN ('para', 'case_manager', 'admin'))
+  role TEXT NOT NULL CHECK (role IN ('para', 'case_manager', 'admin')),
+  email TEXT UNIQUE NOT NULL,
+  email_verified_at TIMESTAMPTZ,
+  image_url TEXT
+);
+
+CREATE TABLE "account" (
+  account_id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
```

Comment:
What is an account_id modeling?

Reply:
sorry, are you asking what the table models or what the specific `account_id` column does?

the table represents the connection between users and third-party SSO providers like Google OAuth

it could technically be one table if you're only ever going to support one SSO provider but I think this is a little cleaner (and what their official adapters do)

Reply:
Thanks - I was asking about what both, the account table, and the account_id field were modeling.

Re user and account table: I imagine that if someone logs in with different SSO providers or even different IDs from the same SSO provider (e.g. different Gmail addresses), they are essentially different users in our system. For now, a simple way to do ACLs is by domain - for e.g. we allow sfusd.edu CM accounts to work with other sfusd.edu para accounts and so on.

Is it a reasonable assumption that each Oauth identity is a separate "user" on Compass? If yes, do we need separate user and account tables?
If not, how will we manage user and account tables?

Reply:
no, NextAuth considers an email to be a user's unique identity

if:
1. someone signed in with Google OAuth using the email `foo@gmail.com`
2. they signed in again with GitHub OAuth, and their GitHub account has the same email `foo@gmail.com`

there would be two rows in `account` (one for Google and one for GitHub) and a single row in `user`

we don't manage any of this ourselves, it's handled by NextAuth through our custom adapter